### PR TITLE
Fix: Wait for the product to be added to cart before redirecting customer to the cart page

### LINKED ISSUE #6390: Caching issue adding items to cart with All Products block
## Describe the bug

When adding items to the cart (it seems to happen mostly when doing so from the main shop page), the item does not appear in the cart until you do a hard refresh.  When using the default WP theme, WooCommerce Blocks and Gutenberg, the issue is there.  If WooCommerce Block is deactivated, the issue does not occur.

## To reproduce
1. Go to WooCommerce > Settings > Products > check both 'add to cart' behaviors to redirect and use AJAX, then save
2. On your main shop page, add the All Products block
3. View shop page in front-end and click "add to cart" for any product
4. Should be redirected to the cart page but the product is not there
5. Make sure to try in multiple browsers with no items in the cart and a clear cache

## Expected behavior

Item should immediately appear in the cart

### WordPress (please complete the following information):

-   Core version: 5.9.3
-   Gutenberg version (if installed): 13.1.0
-   WooCommerce version: 6.4.1
-   WooCommerce Blocks version: 7.5.0
-   Site language: English
-   Any other plugins installed: no

### Desktop (please complete the following information):

-   OS: mac
-   Browser: chrome, safari and ff

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #6390 

This PR fixes the empty cart after adding a product to the cart issue by properly waiting for the product to be added to the cart before redirecting the customer to the cart page. By doing so, the cart will always contain the newly added products.

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) .
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md)
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Go to WooCommerce > Settings > Products > check both 'add to cart' behaviors to redirect and use AJAX, then save
2. On the home page, add the All Products block
3. On the front-end,  click "add to cart" for any product
4. See the page is redirected to the Cart page and the Cart block contains the newly added product.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Fix: All Products block: New product is missing in the Cart block if `Redirect to the cart page after successful addition` is enabled.

### CODE REVIEW

File: assets/js/atomic/blocks/product-elements/button/block.js [hunk 1]
```diff
@@ -147,12 +160,7 @@ const AddToCartButton = ( {
 			dispatchStoreEvent( 'cart-add-item', {
```
Comment (gigitux) [inline assets/js/atomic/blocks/product-elements/button/block.js:147]: I guess that the problem here is that `addToCart` returns a Promise and we are not waiting for it.

I commented the new code, and edited the function `onClick`, in this way:

```
		buttonProps.onClick = async () => {
			await addToCart();
			dispatchStoreEvent( 'cart-add-item', {
				product,
			} );
			const { cartRedirectAfterAdd } = getSetting( 'productsSettings' );
			if ( cartRedirectAfterAdd ) {
				window.location.href = CART_URL;
			}
			setClickedAddToCart( true );
		};
	} 
```

It seems that works. Can you check?

File: src/BlockTypes/AbstractBlock.php [hunk 2]
```diff
@@ -180,7 +180,7 @@ protected function register_chunk_translations( $chunks ) {
 	protected function get_chunks_paths( $chunks_folder ) {
 		$build_path = \Automattic\WooCommerce\Blocks\Package::get_path() . 'build/';
 		$blocks     = [];
-		if ( ! is_dir( $build_path . $chunks_folder ) ) {
+		if ( ! is_dir( $build_path ) || ! is_dir( $build_path . $chunks_folder ) ) {
```
Comment (gigitux) [inline src/BlockTypes/AbstractBlock.php]: Aren't we sure that build_path is a directory?

**Review by gigitux — CHANGES_REQUESTED** [review]
Thanks for working on this PR!

I have some questions. Could you take a look?

File: src/BlockTypes/AbstractBlock.php (hunk 2 above)
Comment (dinhtungdu) [reply src/BlockTypes/AbstractBlock.php]: This is an outdated fix I submitted in #6464. I synced with `trunk`.

File: assets/js/atomic/blocks/product-elements/button/block.js (hunk 1 above)
Comment (dinhtungdu) [reply assets/js/atomic/blocks/product-elements/button/block.js:147]: It's a great solution! Thanks so much! I updated the PR per your suggestion!

**Review by gigitux — APPROVED** [review]
Thanks for addressing comments! LGTM 🚢

**Pull request merged**